From 9f2fe4129503e9d3f22912839b3bf061a01fab0e Mon Sep 17 00:00:00 2001
From: Daniel Leung <daniel.leung@intel.com>
Date: Mon, 11 Aug 2014 17:51:25 -0700
Subject: [PATCH 1/1] input: ft5x06: fix suspend/resume failed messages

It turns out ft5x0x_set_pmode() will return positive numbers
to indicate how many messages are sent. This patch checks for
positive return values from ft5x0x_set_pmode() and sets
the return value of suspend/resume to zero, which the dpm callback
expects for successful operations.

Signed-off-by: Daniel Leung <daniel.leung@intel.com>
---
 drivers/input/touchscreen/ft5x06_ts.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/drivers/input/touchscreen/ft5x06_ts.c b/drivers/input/touchscreen/ft5x06_ts.c
index 824727f..f27885cb 100644
--- a/drivers/input/touchscreen/ft5x06_ts.c
+++ b/drivers/input/touchscreen/ft5x06_ts.c
@@ -778,6 +778,10 @@ static int ft5x0x_ts_suspend(struct device *dev)
 
 	ret = ft5x0x_set_pmode(tsdata, PMODE_HIBERNATE);
 
+	/* (ret > 0) indicates number of messages sent.         */
+	/* Set ret to 0, so not to confuse device pm functions. */
+	if (ret > 0)
+		ret = 0;
 	if (ret < 0)
 		goto out;
 
@@ -807,6 +811,10 @@ static int ft5x0x_ts_resume(struct device *dev)
 
 	ret = ft5x0x_set_pmode(tsdata, PMODE_ACTIVE);
 
+	/* (ret > 0) indicates number of messages sent.         */
+	/* Set ret to 0, so not to confuse device pm functions. */
+	if (ret > 0)
+		ret = 0;
 	if (ret < 0)
 		goto out;
 
-- 
1.8.3.2

